nonce/nonce.go

```diff
@@ -193,3 +221,28 @@ func (ns *NonceService) Valid(nonce string) bool {
 	ns.stats.Inc("Valid", 1)
 	return true
 }
+
+func splitNonce(nonce string) (string, string, error) {
+	if len(nonce) < 4 {
+		return "", "", errInvalidNonceLength
+	}
+	return nonce[:4], nonce[4:], nil
+}
+
+// RemoteRedeem checks the nonce prefix and routes the Redeem RPC
+// to the associated remote nonce service
+func RemoteRedeem(ctx context.Context, noncePrefixMap map[string]noncepb.NonceServiceClient, nonce string) (bool, error) {
```

Comment:
Was there an explicit reason you wanted `RemoteRedeem` to **not** return an error if `splitNonce` returns an err or if there is no matching prefix in `noncePrefixMap`? I feel like these should be error returns but maybe I'm overlooking the reason you avoided that?

Reply:
This is mainly to maintain the behavior that currently exists with the built in nonce service. In the existing code any bad nonce related issues will just result in a false bool that is interpreted as a nonce verification failure and is translated into a BadNonce error. Since RemoteRedeem can return RPC errors we need a bool/error return and I decided to only make the returned errors RPC ones so that we can easily decide if we should return a BadNonce error vs. a InternalServer error.

Reply:
That makes sense, thanks!